connector/github/github.go

```diff
@@ -133,16 +133,22 @@ type githubConnector struct {
 	httpClient *http.Client
 }
 
+// groupsRequired returns whether dex must access GitHub "groups" (org and team)
+// info. If 'orgs' is populated this method returns true regardless of client
+// scope values. However groups scope must be set with 'org' to maintain old behavior.
```

Comment:
Could you add a little reasoning why we "set it true regardless of client scope values"? *since the GitHIb API requires it*